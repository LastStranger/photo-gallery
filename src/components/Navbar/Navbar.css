@import '../../common_styles/colors.css';

.navbar {
  position: fixed;
  bottom: 0;
  background-color: var(--nav-background);
  height: 12vh;
  width: 100%;
  display: flex;
  justify-content: space-around;
  align-items: center;
  opacity: 1;
  transition: opacity .5s ease;

  &.is-invisibile {
    opacity: 0;
    transition: opacity .5s ease;
  }
}

.navbar-item {
  width: 20px;
}
