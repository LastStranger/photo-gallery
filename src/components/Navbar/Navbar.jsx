// @flow

import { h } from 'preact';
import './Navbar.css';

const Navbar = (props: { visibile?: boolean }) =>
  <nav className={`navbar${props.visibile ? '' : ' is-invisibile'}`}>
    <img className="navbar-item" src="/assets/icons/plusOne.jpg" alt="give +1" />
    <img className="navbar-item" src="/assets/icons/comment.jpg" alt="comment" />
    <img className="navbar-item" src="/assets/icons/add.jpg" alt="add" />
    <img className="navbar-item" src="/assets/icons/arrow.jpg" alt="share" />
  </nav>;

Navbar.defaultProps = {
  visibile: true
};

export default Navbar;
