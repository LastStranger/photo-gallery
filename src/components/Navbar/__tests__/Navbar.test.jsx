// @flow

import { h } from 'preact';
import render from 'preact-render-to-string';
import Navbar from '../';

test('renders with visibile', () => {
  const string = render(<Navbar />);
  expect(string).toMatchSnapshot();
});

test('renders without visibile', () => {
  const string = render(<Navbar visibile={false} />);
  expect(string).toMatchSnapshot();
});
