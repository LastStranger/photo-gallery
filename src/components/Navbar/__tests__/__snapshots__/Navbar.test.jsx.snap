// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with visibile 1`] = `
<nav class="navbar">
  <img class="navbar-item" src="/assets/icons/plusOne.jpg" alt="give +1"
  />
  <img class="navbar-item" src="/assets/icons/comment.jpg" alt="comment"
  />
  <img class="navbar-item" src="/assets/icons/add.jpg" alt="add" />
  <img class="navbar-item" src="/assets/icons/arrow.jpg" alt="share" />
</nav>
`;

exports[`renders without visibile 1`] = `
<nav class="navbar is-invisibile">
  <img class="navbar-item" src="/assets/icons/plusOne.jpg" alt="give +1"
  />
  <img class="navbar-item" src="/assets/icons/comment.jpg" alt="comment"
  />
  <img class="navbar-item" src="/assets/icons/add.jpg" alt="add" />
  <img class="navbar-item" src="/assets/icons/arrow.jpg" alt="share" />
</nav>
`;
